resources/analysisTools/copyNumberEstimationWorkflow/beagle_embed_haplotypes_vcf.py

```diff
@@ -0,0 +1,54 @@
+ #!/usr/bin/python
+import argparse
+import gzip
+parser = argparse.ArgumentParser()
+parser.add_argument('--hap_file', help='Input file with phased haplotypes (i.e. output from Beagle)')
+parser.add_argument('--vcf_file', help='Input .vcf file')
+parser.add_argument('--out_file', help='Out .vcf file')
+args = parser.parse_args()
+
+if args.hap_file.split('.')[-1] == 'gz':
+    hap_file = gzip.open( args.hap_file, "r" )
+else:
+    hap_file = open( args.hap_file, "r" )
+vcf_file = open( args.vcf_file, "r" )
+outfile = open( args.out_file, "w" )
+
+hap_line = hap_file.readline()
+while hap_line[0] == "#":
+    hap_line = hap_file.readline()
+
+if hap_line:
+    hap_line = hap_line.rstrip().split()
+
+for vcf_line in vcf_file:
+    
+    if vcf_line[0] != "#":
+        
+        vcf_line = vcf_line.rstrip().split("\t")
+        vcf_line[0] = vcf_line[0].replace('chr', '')
+        vcf_line[0] = vcf_line[0].replace( 'X', '23' )
```

Comment:
What about 'Y'?

Reply:
Good question...

So after checking again with the source code it seems like ACEseq never phases chrY. In `ACESeqMethods.groovy` we have the two functions:
```java
static PhaseGenotypeByChromosome phaseGenotypes(BamFile controlBam) {
        IndexedFileObjects indexedFileObjects = ParallelizationHelper.runParallel(
                COConstants.CVALUE_AUTOSOME_INDICES,
                ACEseqConstants.TOOL_PHASE_GENOTYPES,
                controlBam,
                null,
                ACEseqConstants.PARM_CHR_INDEX,
                getGlobalJobSpecificParameters(controlBam.executionContext.configuration))
        return new PhaseGenotypeByChromosome(indexedFileObjects, controlBam.getExecutionContext())
    }
```
and
```java
static Tuple2<PhasedGenotypeFile, HaploblockGroupFile> phaseGenotypeX(GenderFile sexFile, BamFile controlBam) {
        return (Tuple2<PhasedGenotypeFile, HaploblockGroupFile>) GenericMethod.callGenericTool(ACEseqConstants.TOOL_PHASE_GENOTYPEX, controlBam, sexFile);
    }
```

So only the autosomes and chrX are phased.

The question is now: bug or feature? Maybe @GWarsow can help with this?

Reply:
Ok as discussed in the meeting today, phasing for chrY obviously does not make any sense biologically, so everything is as it should be.